File: tasket/models/rest.py
from dataclasses import dataclass
from typing import Dict, Any


@dataclass
class RestRequest:
    url: str
    url_params: Dict[str, str]
    query_params: Dict[str, str]
    body: Dict[str, Any]
    headers: Dict[str, str]


@dataclass
class RestResponse:
    data: str
    status: int

File: src/rest/rest.go
package rest

type RestRequest struct {
	Url         string
	UrlParams   map[string]interface{}
	QueryParams map[string]interface{}
	Metadata    map[string]interface{}
	Body        []byte
}

type RestError struct {
	Type   string `json:"type"`
	Detail string `json:"detail"`
}

type RestMessage struct {
	Message string `json:"message"`
}

type RestResponse struct {
	Data   interface{}
	Status int64
}
